engine/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStepMeta.java

```diff
@@ -845,9 +844,6 @@ public String buildFilename( VariableSpace space, int stepnr, int splitnr ) {
   @Override
   public void getFields( RowMetaInterface r, String name, RowMetaInterface[] info, StepMeta nextStep,
     VariableSpace space, Repository repository, IMetaStore metaStore ) {
-    if ( r == null ) {
```

Comment:
What happens if you do not initialize the RowMetaInterface? You do not have a NullPointerException at getFields on Step or other action?

Reply:
'r' is a local variable (well, a method parameter) and never used; as so, it was already an unnecessary instantiation.